scripts/checks/verify_supply_chain.py

```
@@ -0,0 +1,158 @@
+    path = Path(".github/dependabot.yml")
+    if not path.exists():
+        return [f"missing file: {path}"]
+    content = path.read_text(encoding="utf-8")
+    missing: list[str] = []
+    for ecosystem in ["npm", "pip", "cargo", "github-actions"]:
+        if f'package-ecosystem: "{ecosystem}"' not in content:
+            missing.append(f"dependabot missing ecosystem: {ecosystem}")
+    return missing
+
+
+def read_workflow(path: Path, label: str, missing: list[str]) -> str:
```

Ruff (0.15.5)

[warning] Unused function argument: `label`

(ARG001)